datastream-kafka/src/test/java/com/linkedin/datastream/kafka/TestKafkaTransportProvider.java

```diff
@@ -281,6 +289,92 @@ public void testSendMultipleEventsInSingleDatastreamProducerRecord() throws Exce
     Assert.assertNotNull(DynamicMetricsManager.getInstance().getMetric(producerCountMetricName));
   }
 
+  private void testEventSendWithTimestamp(int numberOfEvents, int numberOfPartitions, int partition, boolean includeKey,
+                             boolean includeValue, String metricsPrefix, boolean preserveSourceEventTimestamp) throws Exception {
```

Comment:
You pass a boolean preserveSourceEventTimestamp, but I don't see you using it, instead you hard code it to "TRUE" in the test. Can you also add a -ve test for this, i.e. set this to false, and validate that you don't get any source timestamps?

Reply:
added test cases when preserveSourceEventTimestamp is False and when its not configured.